### PR TITLE
build: update to latest rules_typescript

### PR DESCRIPTION
**Pull request by alexeagle**
This changes our devmode to es2015, so we now depend on the full es2015 typings.
Update a spot in the compiler which assumed es5 downleveling.

### CODE REVIEW

File: packages/compiler/src/core.ts [hunk 1]
```diff
@@ -282,16 +282,16 @@ export interface MetadataFactory<T> {
 }
 
 function makeMetadataFactory<T>(name: string, props?: (...args: any[]) => T): MetadataFactory<T> {
-  const factory: any = (...args: any[]) => {
+  function factory(...args: any[]) {
```
Comment (alxhub) [inline packages/compiler/src/core.ts]: Maybe leave a comment here explaining why this is declared as a `function`. Alternatively, maybe the type of `MetadataFactory` could be modified to require a constructable type?

**Review by alxhub — APPROVED** [review]
Approving for the compiler side.

However, please check with @IgorMinar - my understanding was that `es6-subset` exists to limit that surface of ES2015 that we depend on to the bits supported by Closure.

**Comment by alexeagle** [comment]
discussed with @IgorMinar yesterday, he agrees with removing the file. I don't think the original intent was Closure compat, it was to try to prevent Angular depending on a full ES2015 polyfill in old browsers. But now we no longer care about that, since no modern browser needs the polyfills.
It's true that we might now find out about breakages in g3sync which we used to discover when type-checking our code, but OTOH we didn't keep the subset up-to-date and are currently restricted from using some valid ES2015 constructs for no good reason.

**Comment by alfaproject** [comment]
What about IE9?

**Comment by alexeagle** [comment]
@alfaproject IE9 still needs polyfills - in practice I think we've always said to use the full es6-shim. To be precise, I should have said that the thing we no longer care about is making it possible to use a smaller subset of the polyfills with Angular.

**Comment by alfaproject** [comment]
I guess Microsoft doesn't support any IE version except 11 anyway (not even security updates I think).

**Comment by gregmagolan** [comment]
Replaced by #28625

**Pull request closed without merging**